sdk/python/kfp/dsl/_pipeline.py

```diff
@@ -32,7 +33,33 @@ def my_pipeline(a: PipelineParam, b: PipelineParam):
   ```
   """
   def _pipeline(func):
-    Pipeline.add_pipeline(name, description, func)
+    import inspect
+    fullargspec = inspect.getfullargspec(func)
+    args = fullargspec.args
+    annotations = fullargspec.annotations
+
+    # defaults
+    arg_defaults = {}
+    if fullargspec.defaults:
+      for arg, default in zip(reversed(fullargspec.args), reversed(fullargspec.defaults)):
+        arg_defaults[arg] = default
+
+    # Construct the PipelineMeta
+    pipeline_meta = PipelineMeta(name=name, description=description)
+    # Inputs
+    for arg in args:
+      arg_type = TypeMeta()
+      arg_default = arg_defaults[arg] if arg in arg_defaults else ''
```

Comment:
Should we use `None` instead of empty string?

Reply:
Yes. Good idea. I will update that.